cli/src/cli.rs

```diff
@@ -0,0 +1,100 @@
+use {
+    crate::{command::Command, helper::CliHelper, print::Print},
+    gluesql::{
+        prelude::Glue,
+        store::{GStore, GStoreMut},
+    },
+    rustyline::{error::ReadlineError, Editor},
+    std::{
+        fmt::Debug,
+        io::{Result, Write},
+        marker::PhantomData,
+    },
+};
+
+pub struct Cli<T, U, W>
+where
+    T: Debug,
+    U: GStore<T> + GStoreMut<T>,
+    W: Write,
+{
+    glue: Glue<T, U>,
+    print: Print<W>,
+    _marker: PhantomData<T>,
+}
+
+impl<T, U, W> Cli<T, U, W>
+where
+    T: Debug,
+    U: GStore<T> + GStoreMut<T>,
+    W: Write,
+{
+    pub fn new(storage: U, output: W) -> Self {
+        let glue = Glue::new(storage);
+        let print = Print::new(output);
+
+        Self {
+            glue,
+            print,
+            _marker: PhantomData,
+        }
+    }
+
+    pub fn run(&mut self) -> Result<()> {
+        macro_rules! println {
+            ($($p:tt),*) => ( writeln!(&mut self.print.output, $($p),*)?; )
+        }
+
+        self.print.help()?;
+
+        let mut rl = Editor::<CliHelper>::new();
+        rl.set_helper(Some(CliHelper::default()));
+
+        loop {
+            let line = match rl.readline("gluesql> ") {
+                Ok(line) => line,
+                Err(ReadlineError::Interrupted) => {
+                    println!("^C");
+                    continue;
+                }
+                Err(ReadlineError::Eof) => {
+                    println!("bye\n");
+                    break;
+                }
+                Err(e) => {
+                    println!("[unknown error] {:?}", e);
+                    break;
+                }
+            };
+
+            rl.add_history_entry(line.as_str());
+
+            let command = match Command::parse(line.as_str()) {
```

Comment:
nit picking
You can replace `as_str` with `&`.
String also implements AsRef<str>, and if you borrow it, there is no ownership problem.